ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -433,7 +433,10 @@ export const ProfilePage = fp.flow(
 
       const makeProfileInput = ({title, valueKey, isLong = false, ...props}) => {
         let errorText = profile && errors && errors[valueKey];
-        if (valueKey && Array.isArray(valueKey) && valueKey.length > 1) {
+        if (valueKey && !Array.isArray(valueKey)) {
+          valueKey = [valueKey];
```

Comment:
This is interesting, it looks like JS and TS handle spread differently. When the TS compiles down to javascript it must do something to prevent strings from getting split apart on a spread.  The new code looks more correct and in-line with what I would expect. I will put together a code example demonstrating this issue.  What was the specific error?

Reply:
UI unit profile-page test failure:
```
FAIL src/app/pages/profile/profile-page.spec.tsx
  ● ProfilePageComponent › should invalidate inputs correctly

    expect(received).toBeTruthy()

    Received: false

      88 |     const wrapper = component();
      89 |     wrapper.find(TextInput).first().simulate('change', {target: {value: ''}});
    > 90 |     expect(wrapper.find(TextInput).first().prop('invalid')).toBeTruthy();
         |                                                             ^
      91 |   });
      92 | 
      93 |   it('should display/update address', async() => {

      at Object.<anonymous> (src/app/pages/profile/profile-page.spec.tsx:90:61)
```

Reply:
Cool, thanks - it should be possible to set up a more isolated reproducer here (the above failure is highly detached, and required some trial and error to trace down to the cause), I can try and help set this up as well if you'd like.

For the purposes of this PR I think it should be fine to merge now. Pete's comment of "The new code looks more correct and in-line with what I would expect" answers what I was looking for, which is to make sure I wasn't missing something obvious and that the change makes sense.

Reply:
So this is the issue: 

This Typescript code (in version 3.2.z)
```Typescript
const checkValueKey = (valueKey) => {
  console.log(['test', ...valueKey])
}

checkValueKey('test')
```

compiles to this in javascript:
```Javascript
var checkValueKey = function (valueKey) {
    console.log(['test'].concat(valueKey));
};
checkValueKey('test');
```

When using `concat` as a spread operator you will see the behavior we are seeing in the code. The item in the `concat` will not be split apart.

However, the same code when compiled in TS 4.2.3 yields:
```javascript
var __spreadArray = (this && this.__spreadArray) || function (to, from) {
    for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
        to[j] = from[i];
    return to;
};
var checkValueKey = function (valueKey) {
    console.log(__spreadArray(['test'], valueKey));
};
checkValueKey('test');
```
Where the `__spreadArray` function more closely follows what JS does with the spread operator, splitting apart the string.

Reply:
Nice find! Thanks. That seems like the likely culprit.

Something that's still not obvious to me though is why the typescript version changed with this PR, that seems like an unexpected side effect. My best guess is that changing `ts-jest` changed the Typescript version behind the scenes for tests, but if so this interaction is not obvious from `yarn.lock` or from the ts-jest docs.

Reply:
One more update - I was wondering what our ES target version was, and whether a change to that would prevent substituting the spread operator at all, I don't think there is any browser we are targeting that doesn't support this.
Looking at the `tsconfig.json` L13 we are changing our ES target to `es6` in this change.
Prior to this change, the code in question would compile to (note the concat) :

`return _this.setState(lodash_fp__WEBPACK_IMPORTED_MODULE_1__["set"](['currentProfile'].concat(valueKey), v));` 

and it now compiles to:

`onChange: v => this.setState(lodash_fp__WEBPACK_IMPORTED_MODULE_1__["set"](['currentProfile', ...valueKey], v)),`

I am curious as to whether the error popped up as a result of the target change, or the jest upgrade. In other words - is jest using a different typescript version from our build?

Reply:
Alright... Final update... just for a full understanding of what is happening.
It looks like ts-jest **is** using our version of TS, the real issue is with the `target` change in the config and how it effects the resulting Javascript that comes out.

This is what I tested to confirm the behavior of the tests:
```
ts-jest version  | target | result
----------------------------------------------
25.3.0           | es5    |  pass (because it uses concat )
25.3.0           | es6    |  fail (uses spread operator)
26.5.4           | es5    |  pass (uses concat) --> If this were using 4.x of TS it would fail by using `__spreadArray`
26.5.4           | es6    |  fail (uses spread operator)
```

Reply:
I'm not 100% confident but it looks to me that there was one usage of valueKey that this missed: 
```valueKey: 'verifiedInstitutionalAffiliation.institutionDisplayName'``` -> https://github.com/all-of-us/workbench/pull/4822